src/CMakeLists.txt

```diff
@@ -269,16 +270,22 @@ elseif(${CMAKE_Fortran_COMPILER_ID} STREQUAL Cray )
    endif()
 elseif(${CMAKE_Fortran_COMPILER_ID} STREQUAL NVHPC )
    target_compile_definitions(${PROJECT_NAME} PRIVATE ENABLE_NVHPC_WORKAROUNDS)
-   target_compile_options(${PROJECT_NAME} PRIVATE -fast -fastsse -O3 -Mallocatable=95 -Mr8 -pgf90libs)
-   if(ENABLE_OPENACC)
+   #target_compile_options(${PROJECT_NAME} PRIVATE -fast -fastsse -O3 -Mallocatable=95 -Mr8 -pgf90libs)
+   target_compile_options(${PROJECT_NAME} PRIVATE -Mnofma -Mallocatable=95 -Mr8 -pgf90libs)
+   if(${ENABLE_OPENACC})
       # additional compiler settings
-      target_compile_options(${PROJECT_NAME} PRIVATE -acc -ta=tesla:${NV_GPU_ARCH} -Minfo=accel)
-      set(CMAKE_EXE_LINKER_FLAGS "-acc -ta=tesla:${NV_GPU_ARCH}")
+      message("Taking ENABLE_OPENACC = ON")
+      target_compile_options(${PROJECT_NAME} PRIVATE -acc -O2 -gpu=${GPU_FLAGS} -Minfo=accel)
+      # set(CMAKE_EXE_LINKER_FLAGS "-acc -gpu=${GPU_FLAGS}")
+      if(${DISABLE_OPENACC_ATOMICS})
+         message("Taking DISABLE_OPENACC_ATOMICS = ON")
+          target_compile_definitions(${PROJECT_NAME} PRIVATE DISABLE_OPENACC_ATOMICS)
+      endif()
    endif()
-   if(ENABLE_OPENMP)
+   if(${ENABLE_OPENMP})
       target_compile_options(${PROJECT_NAME} PRIVATE -Mipa=fast)
-   else()
-      target_compile_options(${PROJECT_NAME} PRIVATE -Mipa=fast,inline)
+   # else()
+   #    target_compile_options(${PROJECT_NAME} PRIVATE -Mipa=fast,inline)
```

Comment:
@basava70, do we need these lines commented out for things to work for FESOM with GPUs? If we don't need them commented out please bring them back in.

Reply:
Seems like we dont need these commented out.